### ISSUE TITLE
generic guitar link

### ISSUE THREAD

**Issue by AdalbertoFranco1999** [mdias/rs_asio#176]
Is it possible to use the generic guitarlink? as a connection? I have some difficulties.
If not, is there a plan to work?

my guitar link: 
https://pt.aliexpress.com/item/4001266508171.html?src=google&albch=shopping&acnt=494-037-6276&isdl=y&slnk=&plac=&mtctp=&albbt=Google_7_shopping&aff_platform=google&aff_short_key=UneMJZVf&&albagn=888888&isSmbAutoCall=false&needSmbHouyi=false&albcp=1718811104&albag=67397419396&trgt=1074811600784&crea=pt4001266508171&netw=u&device=c&albpg=1074811600784&albpd=pt4001266508171&gclid=CjwKCAjwxuuCBhATEiwAIIIz0dTqTO0ldPfgI2OuP7HmNpnt7tMVT6eOs_AfnhEBRf_Vf4yanYPNGBoCJo8QAvD_BwE&gclsrc=aw.ds

**Comment by zagatozee** [comment]
If you can find ASIO drivers for that, then yeah - you could, but really, just using the Direct Connect mode option is almost certainly the easier option - for that interface.

You can add the DC mode with another project here on github - RSMods - https://bit.ly/RSModsGit

An out of date features wise, write up on some of the things added in RSMods; https://theriffrepeater.com/rsmods-but-what-is-direct-connect-mode-mods-pc-mac-steam-custom-tuning-setlist-cables-interfaces/

**Comment by AdalbertoFranco1999** [comment]
I defined the guitarlink as a microphone, it worked very well, but, the game does not leave the audio of the instrument, with this mod is it possible to play and listen? put mod generates BAN? (I made the guitarlink work like the real tone cable, but the sound was terrible)

**Comment by zagatozee** [comment]
Direct Connect mode is literally just the microphone mode, but with tone emulations enabled - so yes. If you got detection working in MM, then DC will be fine. 

DC mode is actually a game mode made by UbiSoft, that was disabled (hidden) in the release version, for reasons we are not sure of. We just re-enabled it and it works fine for simple interfaces.

**Comment by AdalbertoFranco1999** [comment]
Okay, I'll try, anything I send a message. I'm thinking of buying a usb-p10 cable or a dedicated audio system to play and study. do you think i can get a better result with a usb / p10 cable? or will the sound get bad? the original cable is extremely expensive in the country I live in and the freight to buy in the USA would give xomprar to buy 3 original cables hahahaha thanks for listening

**Comment by zagatozee** [comment]
I'm in a similar boat, but was fortunately able to find the RTC once locally for a mildly offensive price, rather than having to sell a kidney.  
  
I've tested DC mode with all sorts of devices. As long as it is capable fo being set to 48khz 16bit dvd quality in windows, it'll work with DC mode.

**Comment by mdias** [comment]
The official RealTone Cable will give you good results in Rocksmith, as well as lower headache to set it up (there's no setup needed).
Other USB to Guitar cables will need you to fiddle with them to work in Rocksmith, and it's more difficult to be sure about their quality.

If you plan on using the guitar outside of Rocksmith but still on your computer you'll most likely want something with ASIO drivers as most guitar apps will expect that and you're more likely to get a better experience with them. There is ASIO emulation software like ASIO4ALL that emulates the existence of ASIO drivers, but most of the time they're not on the same level as proper ASIO drivers.

So to sum it up:

- Easiest solid experience in Rocksmith:
Use the RTC
- Cheaper than RTC but usually not better:
The DirectConnect mode should work with most interfaces/cables, but no guarantees!
- Serious about guitar even outside of Rocksmith:
Buy an actual audio interface with proper ASIO drivers; also not guaranteed to work flawlessly with Rocksmith, but between DC mode and RS_ASIO there's a very low chance you won't be able to play


> As long as it is capable fo being set to 48khz 16bit dvd quality in windows, it'll work with DC mode.

Also follow this advice, as it's **very** important for Rocksmith.

**Comment by AdalbertoFranco1999** [comment]
Got it friend, thanks for your attention
last doubt hahaha
I put the direct connection and the audio is not emulated, the popped sound of the interface continues

**Comment by zagatozee** [comment]
The popping sounds means the buffer settings in your Rocksmith.ini file, are not good for your combination of device and CPU. There is a PDF file in the rocksmith folder that can help you tweak those settings.
